### PR TITLE
WebSockets tutorial

### PR DESCRIPTION
**Pull request by thegoofist**
I ended up noticing some activity on the [Woo repo](https://github.com/fukamachi/woo/pull/81) that lead me to believe my initial trouble wasn't with Clack+websocket-driver but was instead with Woo, so that the issues raise din #255 are probably not relevant. 

Hence, I have rewritten the tutorial to use Clack and websocket-driver !

### CODE REVIEW

File: websockets.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+~~~
+
+
+## The websocket-driver Concept
+
+In websocket-driver, a WebSocket connection is an instance of the `ws` class,
+which exposes an event-driven API. You register event handlers by passing your
+WebSocket instance the method `on`, e.g. `(on :message my-websocket #'some-message-handler)`, 
+where `some-message-handler` would be evoked whenever a new message arrives.
```
Comment (mohe2015) [inline websockets.md]:
```suggestion
where `some-message-handler` would be invoked whenever a new message arrives.
```
Warning: English is not my native language

File: websockets.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+The `websocket-driver` API provides for handlers on the following events:
```
Comment (mohe2015) [inline websockets.md]:
```suggestion
The `websocket-driver` API provides handlers for the following events:
```
Warning: English is not my native language

File: websockets.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+- `:open`: When a connection is opened. Expects a thunk as its handler.
```
Comment (mohe2015) [inline websockets.md]: What is a thunk? Maybe you should replace that word?

File: websockets.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+- `:message` When a message arrives. Expects a handler of one argument, the message received.
```
Comment (mohe2015) [inline websockets.md]:
```suggestion
- `:message` When a message arrives. Expects a handler with one argument, the message received.
```
Warning: English is not my native language

File: websockets.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+- `:close` When a connection closes. Expects a handler with two keyword args, a
+  "code" and a "reason" for the dropped connection.
+- `:error` When some kind of protocol level error occurs. Expects a handler of
```
Comment (mohe2015) [inline websockets.md]:
```suggestion
- `:error` When some kind of protocol level error occurs. Expects a handler with
```
Warning: English is not my native language

File: websockets.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  one argument, the error message. 
+  
+For the purposes of your chat server, you will want to handle the case when a
+new user arrives to the channel, when the user sends a message to the channel,
```
Comment (mohe2015) [inline websockets.md]:
```suggestion
new user arrives to the channel, when a user sends a message to the channel,
```
Warning: English is not my native language

File: websockets.md (hunk 3 above)
Comment (thegoofist) [reply websockets.md]: Ah sure. Sorry a thunk is a function of zero arguments. I can just say that.

**Comment by thegoofist** [comment]
@mohe2015  Thanks for the review!

**Comment by pnathan** [comment]
Spiffy! However, issue #255 mentions *not* using `websocker-driver`, but you are, here?

**Comment by thegoofist** [comment]
@pnathan That's correct, see my comment at the top of this PR. 

I had initially wanted to use Clack and websocket-driver, using Woo as the underlying web server, but had some issues. I later discovered that the issues had to do with Woo and were not directly related to Clack and websocket-driver. 

And so, because the community seemed to prefer the Clack ecosystem, and because I had wanted to use Clack initially anyway, I re-wrote it to use Clack and websocket-driver.

**Comment by pnathan** [comment]
@thegoofist cool!

**Comment by pnathan** [comment]
hm @thegoofist can you put in another PR to add websockets.md to the index.md link list? I should have caught that, I had forgotten we didn't autogen our index page

**Comment by thegoofist** [comment]
Ack. Oh right. Np. Just a sec.

**Comment by vindarel** [comment]
Very useful and welcome, thanks !

about quoting simple quotes: this library gives pythonic triple-quotes: https://github.com/smithzvk/pythonic-string-reader

FYI I posted the tutorial on reddit, we might have feedback: https://www.reddit.com/r/Common_Lisp/comments/ctyzz7/websockets_tutorial_with_websocketdriver_the/

**Pull request merged**